### ISSUE TITLE
reconcile error while use K8s Service as trait

### ISSUE THREAD

**Issue by wonderflow** [oam-dev/kubevela#876]
It report errors:

```
Cannot apply components	{"controller": "oam/applicationconfiguration.core.oam.dev", "request": "default/application-sample", "uid": "6e97f944-e107-47ef-ba7b-bf686f46e554", "version": "726180409", "error": "cannot apply trait \"v1\" \"Service\" \"myweb\": cannot update object: Service \"myweb\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable", "errorVerbose": "Service \"myweb\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable
```

Reproduce yaml:

```
apiVersion: core.oam.dev/v1alpha2
kind: Component
metadata:
  name: myweb
  namespace: default
spec:
  workload:
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        workload.oam.dev/type: worker
    spec:
      replicas: 10
      selector:
        matchLabels:
          app.oam.dev/component: myweb
      template:
        metadata:
          labels:
            app: myweb
            app.oam.dev/component: myweb
        spec:
          containers:
          - command:
            - sleep
            - "1000"
            image: busybox
            name: myweb
          - image: nginx
            name: sidecar-test
---
apiVersion: core.oam.dev/v1alpha2
kind: ApplicationConfiguration
metadata:
  name: application-sample
  namespace: default
spec:
  components:
  - componentName: myweb
    traits:
    - trait:
        apiVersion: v1
        kind: Service
        metadata:
          labels:
            trait.oam.dev/type: kservice
          name: myweb
        spec:
          ports:
          - port: 80
            targetPort: 80
          selector:
            app: myweb
```

The controller keep reconcile and update the object. While the underlying K8s service spec has been changed:

The fields `clusterIP` and `type` were automatically added. So it complains if the kubevela controller want to change it.

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.oam.dev/generation: "1"
  creationTimestamp: "2021-01-11T09:44:56Z"
  labels:
    app.oam.dev/component: myweb
    app.oam.dev/name: application-sample
    app.oam.dev/resourceType: TRAIT
    app.oam.dev/revision: ""
    application.oam.dev: application-sample
    trait.oam.dev/type: kservice
  name: myweb
  namespace: default
spec:
  clusterIP: 172.21.7.149
  ports:
  - port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: myweb
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

**Comment by resouer** [comment]
Let's revisit when reconcile on update is needed in the context of `Application` object is now online. IMO, only fields defined in Application (i.e. parameters in Definition) needs to be enforced by OAM controller.

**Comment by captainroy-hy** [comment]
Having a test in my local integrated env, #857 can fix this issue. No error occurs during reconciling appconfig.

**Comment by wonderflow** [comment]
temporarily fixed by #857
